src/IntegrationTests/Applications/SampleAspNetCoreApp/Controllers/WeatherForecastController.cs

```diff
@@ -19,6 +28,21 @@ public class WeatherForecastController : ControllerBase
         [HttpGet]
         public IEnumerable<WeatherForecast> Get()
         {
+            var metricBuilder = MetricBuilder.CreateCountMetric("WeatherForecast/Get")
+            .WithTimestamp(DateTime.Now)
+            .WithValue(1)
+            .WithIntervalMs(10)
+            .WithAttribute("testAttributeKey1", "testAttributeValue1");
+            
+            var metric = metricBuilder.Build();
+
+            var metricBatch = MetricBatchBuilder.Create()
+            .WithMetric(metric)
+            .Build();
+
+            var dataSender = new MetricDataSender(_config);
```

Comment:
How do you feel about using Dependency Injection to inject this into the controller but resolve it in the container as a singleton.?